### DISCUSSION TITLE
[5.1.1][4.4.5] Wrong URL from menu category / filter tag - MULTILANGUAGE

### DISCUSSION THREAD

**Discussion by mrownicki** [joomla/joomla-cms#47041] [2. Cold Features]
### Steps to reproduce the issue
I think I found a bug in the URL generation. Menu/category/category-tag
If we have a category in menu, and we choose this same category and then filter by TAG 
Site must turn ON **MULTILANGUAGE**
And we have a list of articles, the URL is always with the last tag and not filtered url correct.
Category is person
Menu look like:
Main category - Single Artcle
Subcategory - person 
subcategory - A (Category person filter tag A)
...
subcategory - B (Category person filter tag Z)
Example wrong url:
.../en/person/**tag-z/a-nickname**
.../en/person/**tag-z/k-nickname**

All article have url in last tag from menu this case "z" 
You can call up all articles from each TAGs from A - Z

I can manually call up the correct address
.../en/person/**tag-a/a-nickname**
.../en/person/**tag-k/k-nickname**

I have added the whole alphabet to the menu.
But the url always takes the last menu with the tag example "z"

Of course, all articles have only one tag
Structure tag

Index
-A
...
-Z

### Expected result

Correct URL
.../en/person/tag-a/a-nickname
.../en/person/tag-k/k-nickname

### Actual result
Multilanguage on generate wrong urls
Wrong URL
.../en/person/tag-z/a-nickname
.../en/person/tag-z/k-nickname
### System information (as much as possible)
Joomla 4.4.5

Joomla 5.1.1

PHP 8.2

### Additional comments
The website will finally be updated to Joomla 5 
I haven't had time to verify whether the 5th version is identical, but I suspect it is.
I would rather use tags than multiply unnecessary categories from the alphabet...
UPDATE:
video for better explanation.
https://drive.google.com/file/d/15_2wX66Yrfwb8ASfWbIwYJq3IerpYc2T/view?usp=sharing
After update to 5.1.1 bug still exist.

**Comment by mrownicki** [comment]
@Hackwar do you look and help?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43523">issues.joomla.org/tracker/joomla-cms/43523</a>.</sub>

**Comment by Hackwar** [comment]
The router does not support filtering by tag in a com_content view. Changing this requires a major rewrite and will most likely not happen anytime soon.

**Comment by mrownicki** [comment]
@Hackwar I don't know if we understand each other well.
With multilingualism turned off. 
**Address is correct.**
One language works good.
Two languages not. 
I make video for more explanation.
https://drive.google.com/file/d/15_2wX66Yrfwb8ASfWbIwYJq3IerpYc2T/view?usp=sharing

I can share backup if you want. 
Regards.

**Comment by mrownicki** [comment]
@Hackwar Is this scheduled for fix? Wi have Joomla 6 and coming 6.1 <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43523">issues.joomla.org/tracker/joomla-cms/43523</a>.</sub>
